### PR TITLE
Adds the transform module

### PR DESCRIPTION
**Pull request by funkyfuture**
here's the last addition that i'd like to include w/ the `0.3` release.

once merged, i'll release the probably last beta of it.

### CODE REVIEW

File: delb/transform.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+Let's look at it with examples::
+
+   from delb.transform import Transformation
+
+
+   class ResolveCopyOf(Transformation):
+       def transform(self):
+           for node in self.root.css_select("*[copyOf]"):
+               source_id = node["copyOf"]
+               source_node = delb.document.xpath(
```
Comment (JKatzwinkel) [inline delb/transform.py]:
```suggestion
               source_node = self.origin_document.xpath(
```

File: delb/transform.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                   f'//*[@xml:id="{source_id[1:]}"]'
```
Comment (JKatzwinkel) [inline delb/transform.py:15]: this throws an Exception in `collections.UserString#__getitem__`. What is the slice supposed to mean!?

File: tests/test_transform.py [hunk 3]
```diff
@@ -52,16 +55,35 @@ def transform(self):
...
+    )
+
+
+def test_transformation_sequence():
     document = Document(
         """\
-        <root>
-            <div id="1">
+        <root xmlns:xml="{}">
+            <div xml:id="d1">
```
Comment (funkyfuture) [inline tests/test_transform.py:81]: yeah sure. :1st_place_medal:

File: tests/test_transform.py [hunk 4]
```diff
@@ -52,16 +55,35 @@ def transform(self):
...
+        '<root><b><c>hi</c></b><a>na?</a></root>'
+    )
+
+
+def test_transformation_sequence():
     document = Document(
         """\
-        <root>
-            <div id="1">
+        <root xmlns:xml="{}">
```
Comment (funkyfuture) [inline tests/test_transform.py:80]: this really should not be necessary.

**Review by JKatzwinkel — DISMISSED** [review]

**Comment by JKatzwinkel** [comment]
I'd like to merge https://github.com/funkyfuture/delb/pull/24 because of Python 3.6 and that `black` issue.

**Comment by funkyfuture** [comment]
can you please check whether i oversaw something?

i find your test transformation a little complicated, but they are a good example for sparse, target-hitting code.

**Review by JKatzwinkel — APPROVED** [review]

**Pull request merged**